k8s/plugin/src/resources/upgrade.rs

```diff
@@ -433,38 +476,83 @@ impl UpgradeResources {
     pub async fn get_cluster_role(&self) -> Result<ObjectList<ClusterRole>, Error> {
-        let lp = ListParams::default()
-            .fields(&format!("metadata.name={}", UPGRADE_OPERATOR_CLUSTER_ROLE));
+        let lp = ListParams::default().fields(&format!(
+            "metadata.name={}",
+            upgrade_group(self.release_name.clone(), UPGRADE_OPERATOR_CLUSTER_ROLE)
+        ));
         Ok(self.cluster_role.list(&lp).await?)
     }
 
     /// Return results as list of cluster role binding.
     pub async fn get_cluster_role_binding(&self) -> Result<ObjectList<ClusterRoleBinding>, Error> {
         let lp = ListParams::default().fields(&format!(
             "metadata.name={}",
-            UPGRADE_OPERATOR_CLUSTER_ROLE_BINDING
+            &upgrade_group(
+                self.release_name.clone(),
+                UPGRADE_OPERATOR_CLUSTER_ROLE_BINDING
+            )
         ));
         Ok(self.cluster_role_binding.list(&lp).await?)
     }
 
     /// Return results as list of deployments.
     pub async fn get_deployment(&self) -> Result<ObjectList<Deployment>, Error> {
-        let lp = ListParams::default()
-            .fields(&format!("metadata.name={}", UPGRADE_CONTROLLER_DEPLOYMENT));
+        let lp = ListParams::default().fields(&format!(
+            "metadata.name={}",
+            &upgrade_group(self.release_name.clone(), UPGRADE_CONTROLLER_DEPLOYMENT)
+        ));
         Ok(self.deployment.list(&lp).await?)
     }
 
     /// Return results as list of service.
     pub async fn get_service(&self) -> Result<ObjectList<Service>, Error> {
-        let lp =
-            ListParams::default().fields(&format!("metadata.name={}", UPGRADE_OPERATOR_SERVICE));
+        let lp = ListParams::default().fields(&format!(
+            "metadata.name={}",
+            &upgrade_group(self.release_name.clone(), UPGRADE_OPERATOR_SERVICE)
+        ));
         Ok(self.service.list(&lp).await?)
     }
 }
+
+/// Return results as list of deployments.
+pub async fn get_deployment_for_rest(ns: &str) -> Result<ObjectList<Deployment>, Error> {
+    let client = Client::try_default().await?;
+    let deployment = Api::<Deployment>::namespaced(client.clone(), ns);
+    let lp = ListParams::default().labels(API_REST_LABEL_SELECTOR);
+    Ok(deployment.list(&lp).await?)
+}
+
+pub async fn get_release_name(ns: &str) -> Result<String, Error> {
+    let mut release_name = DEFAULT_RELEASE_NAME.to_string();
+    match get_deployment_for_rest(ns).await {
+        Ok(deployment) => {
+            if deployment.iter().count() == 1 {
+                release_name = match &deployment.items[0].metadata.labels {
+                    Some(label) => match label.get("helm-release") {
+                        Some(value) => value.to_string(),
+                        None => DEFAULT_RELEASE_NAME.to_string(),
+                    },
+                    None => DEFAULT_RELEASE_NAME.to_string(),
+                };
+            } else {
+                println!(
```

Comment:
What is the logic here, if we can't find the deployment we use the default name? But aren't we already in an error condition, is it worth proceeding?
Oh also we may panic down below if count is 0 ? try and use .get(index) as much as possible instead of accessing directly as it may panic

Reply:
Login changed :